src/main/java/edu/cornell/kfs/fp/batch/service/impl/CreateAccountingDocumentServiceImpl.java

```diff
@@ -118,14 +155,64 @@ protected void processAccountingDocumentFromXml(String fileName, CreateAccountin
+            logReport.getExcludedDuplicateFiles().add(fileName);
+            logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
+        } else if (reportItem.isNonBusinessRuleFailure()) {
             logReport.getFilesWithNonBusinessRuleFailures().add(fileName);
         } else {
             logReport.getFilesSuccessfullyProcessed().add(fileName);
         }
         LOG.info("processAccountingDocumentFromXml: Value of reportItem.isNonBusinessRuleFailure just prior to method return =" + reportItem.isNonBusinessRuleFailure());
     }
-    
+
+    protected CreateAccountingDocumentFileEntry findExistingFileEntry(String fileName) {
+        String cleanedFileName = getCleanedFileName(fileName);
+        return businessObjectService.findBySinglePrimaryKey(CreateAccountingDocumentFileEntry.class, cleanedFileName);
+    }
+
+    protected String getCleanedFileName(String fileName) {
+        String cleanedFileName = fileName;
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.BACKSLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.BACKSLASH);
+        }
+        if (StringUtils.contains(cleanedFileName, CUKFSConstants.SLASH)) {
+            cleanedFileName = StringUtils.substringAfterLast(cleanedFileName, CUKFSConstants.SLASH);
+        }
+        cleanedFileName = StringUtils.lowerCase(cleanedFileName, Locale.US);
+        return cleanedFileName;
+    }
+
+    private boolean shouldExcludePreviouslyProcessedFiles() {
+        return parameterService.getParameterValueAsBoolean(KFSConstants.CoreModuleNamespaces.FINANCIAL, 
+                CuFPParameterConstants.CreateAccountingDocumentService.CREATE_ACCOUNTING_DOCUMENT_SERVICE_COMPONENT_NAME, 
+                CuFPParameterConstants.CreateAccountingDocumentService.DUPLICATE_FILE_CHECK_IND);
+    }
+
+    private AccountingXmlDocumentListWrapper unmarshalAccountingDocumentFile(String fileName) {
+        byte[] fileData = LoadFileUtils.safelyLoadFileBytes(fileName);
+        return (AccountingXmlDocumentListWrapper) batchInputFileService.parse(
+                accountingDocumentBatchInputFileType, fileData);
+    }
+
+    private void createOrUpdateFileEntry(AccountingXmlDocumentListWrapper accountingXmlDocuments,
+            CreateAccountingDocumentFileEntry existingEntry, String fileName) {
+        CreateAccountingDocumentFileEntry fileEntry;
+        if (ObjectUtils.isNull(existingEntry)) {
+            fileEntry = new CreateAccountingDocumentFileEntry();
+            fileEntry.setFileName(getCleanedFileName(fileName));
+        } else {
+            fileEntry = existingEntry;
+        }
+        fileEntry.setFileCreatedDate(new Timestamp(accountingXmlDocuments.getCreateDate().getTime()));
+        fileEntry.setFileProcessedDate(dateTimeService.getCurrentTimestamp());
+        fileEntry.setReportEmailAddress(accountingXmlDocuments.getReportEmail());

```

Comment:
similar to the length check on the overview, I think we want a length check on the email and file name

Reply:
I'll add the extra length checks accordingly.